collect_app/src/androidTest/java/org/odk/collect/android/feature/formentry/RequiredQuestionTest.kt

```diff
@@ -0,0 +1,134 @@
+package org.odk.collect.android.feature.formentry
+
+import androidx.test.ext.junit.runners.AndroidJUnit4
+import org.hamcrest.MatcherAssert.assertThat
+import org.hamcrest.Matchers.equalTo
+import org.junit.Rule
+import org.junit.Test
+import org.junit.rules.RuleChain
+import org.junit.runner.RunWith
+import org.odk.collect.android.support.StorageUtils.getAuditLogForFirstInstance
+import org.odk.collect.android.support.rules.CollectTestRule
+import org.odk.collect.android.support.rules.TestRuleChain.chain
+import org.odk.collect.strings.R
+
+@RunWith(AndroidJUnit4::class)
+class RequiredQuestionTest {
+    var rule: CollectTestRule = CollectTestRule()
+
+    @get:Rule
+    var ruleChain: RuleChain = chain()
+        .around(rule)
+
+    @Test
+    fun `requiredQuestionIsMarkedWithAnAsterisk`() {
+        rule.startAtMainMenu()
+            .copyForm("required_question_with_custom_error_message.xml")
+            .startBlankForm("required_question_with_custom_error_message")
+            .assertText("* Required question")
```

Comment:
We've already got a helper for this assertion: you can use `assertQuestion("Required question", isRequired = true)`. You can do the same with `swipeToNextQuestion` as well.